R/SupervisedTask.R

```diff
@@ -1,13 +1,13 @@
-makeSupervisedTask = function(type, data, target, weights, blocking, fixup.data, check.data) {
-  task = makeTask(type = type, data = data, weights = weights, blocking = blocking, fixup.data = fixup.data, check.data = check.data)
+makeSupervisedTask = function(type, data, target, weights, blocking, fixup.data, check.data, spatial) {
+  task = makeTask(type = type, data = data, weights = weights, blocking = blocking, fixup.data = fixup.data, check.data = check.data, spatial = spatial)
 
   if (check.data) {
     # costsens does not have a target col...
     # assertCharacter(target, any.missing = FALSE, min.len = 1L)
     w = which.first(target %nin% colnames(data))
     if (length(w) > 0L)
       stopf("Column names of data doesn't contain target var: %s", target[w])
-    checkTaskData(task$env$data, cols = setdiff(colnames(data), target))
+    checkTaskData(task$env$data, cols = setdiff(colnames(task$env$data), target))
```

Comment:
> modifying the underlying data without telling the user is ever the right way.

Well, we tell beforehand in the help file of `Task` and in the (not yet written) tutorial. 

Implementing so many manual subsetting steps would really cause a lot of new + duplicate code and is error prone. I would still vote against it atm. Maybe we can have a third opinion here?

I could try to display it more prominently that `x` and `y` will only be used for partitioning and not used during modeling? Maybe a info message during task creation? 

> For example, can you build models multiple times on the same task? Once x and y are deleted they aren't put back.

the task itself stays the same, its only modified during `resample` or `train`. Or do you mean 'stacking' when talking about multiple models?